_source/_posts/2022-09-08-react-fastify-postgres.md

```diff
@@ -0,0 +1,917 @@
+```yml
+PORT=3001
+REACT_APP_OKTA_ISSUER=https://{yourOktaDomain}/oauth2/default
+REACT_APP_OKTA_CLIENT_ID={yourOktaClientId}
+REACT_APP_OKTA_BASE_REDIRECT_URI=http://localhost:3001
+```
+
+**NOTE:** We are switching the default port React will run on because the backend API app will use port `3000` by default as well.
+
+### Create React components
+
+Let's create our React components for render on the frontend.
+
+1. In a newly created `src/components` directory, add the following files:
+
+#### login.tsx
+
+```tsx
+import { useOktaAuth } from "@okta/okta-react";
+import { Navigate, useNavigate } from "react-router-dom";
+import "../App.css";
+
+export function Login() {
+  const navigate = useNavigate();
+  const { authState } = useOktaAuth();
+
+  const handleLoginClick = () => {
+    navigate("/facilities");
+  };
+
+  return authState?.isAuthenticated ? (
+    <Navigate to="/facilities" replace />
+  ) : (
+    <div className="form-wrapper">
+      <form onSubmit={handleLoginClick}>
+        <h2>Welcome Back!</h2>
+        <input type="submit" value="Login" />
+      </form>
+    </div>
+  );
+}
+
+```
+
+**NOTE:** Our app uses [React Router v6](https://reactrouter.com/en/main), which utilizes the new `useNavigate` to replace `useHistory` to programmatically navigate.
+
+#### facilities.tsx
+
+```tsx
+import { useOktaAuth } from "@okta/okta-react";
+import { useEffect, useState } from "react";
+import { getErrorMessage } from "../utils/get-error-message";
+
+interface IFacility {
+  Center: string;
+  Facility: string;
+  Status: string;
+  City: string;
+  State: string;
+  Visited: boolean;
+  id: bigint;
+}
+
+function Facilities() {
+  const [data, setData] = useState<IFacility[]>();
+  const [errors, setErrors] = useState<string>();
+  const { authState, oktaAuth } = useOktaAuth();
+
+  const logout = async () => {
+    try {
+      await oktaAuth.signOut();
+    } catch (err) {
+      throw err;
+    }
+  };
+
+  useEffect(() => {
+    const apiCall = async () => {
+      if (authState?.isAuthenticated && authState.accessToken?.accessToken) {
+        try {
+          const response = await fetch("/facilities", {
+            headers: {
+              Authorization: authState.accessToken.accessToken,
+            },
+          });
+          const data = await response.json();
+          setData(data);
+        } catch (error: unknown) {
+          setErrors(getErrorMessage(error));
+        }
+      }
+    };
+    apiCall();
+  }, [authState]);
+
+  const handleVisitedClick = (
+    e: React.ChangeEvent<HTMLInputElement>,
+    facilityId: bigint
+  ) => {
+    const url = `/facilities/${facilityId}`;
+
+    const apiCall = async () => {
+      if (authState?.isAuthenticated && authState.accessToken?.accessToken) {
+        try {
+          await fetch(url, {
+            method: "PATCH",
+            headers: {
+              Authorization: authState.accessToken?.accessToken,
```

Comment:
Ok, yeah. All the places where the Authorization header is set should include Bearer token.
`Authorization: Bearer <accesstokenblah>`